src/ObjectGraph.es6.js

```diff
@@ -181,9 +182,34 @@ ObjectGraph.prototype.rewriteName = function(name) {
 
 // Visit the prototype of o, given its dataMap.
 ObjectGraph.prototype.visitPrototype = function(o, dataMap) {
-  this.storeProto(uid.getId(o), this.visitObject(o.__proto__));
+  this.storeProto(uid.getId(o), this.visitObject(o.__proto__, {proto: true}));
 };
 
+// getProtoPropertyNames returns all inherited properties
+// that have exception values in their prototypes.
+ObjectGraph.prototype.getProtoPropertyNames = function(id) {
+  var protoId = this.getPrototype(id);
+  if (protoId && protoId >= this.root) {
```

Comment:
Please do not use `>= this.root`. We should rely on no such guarantee. I think a viable replacement is `!this.isType(protoId)`.